component/src/test/java/org/wso2/extension/siddhi/map/binary/BinarySinkMapperTestCase.java

```diff
@@ -197,6 +198,90 @@ public String getTopic() {
+        String query = ("@info(name = 'query1') " +
+                "from inputStream " +
+                "select *  " +
+                "insert into outputStream;");
+
+        siddhiManager.createSiddhiAppRuntime(inStreamDefinition + query);
+    }
+
+    @Test
+    public void binarySinkMapperTest4() throws InterruptedException {
+        LOG.info("binary SinkMapper TestCase for single event");
+        SiddhiManager siddhiManager = new SiddhiManager();
+
+        String inStreamDefinition = "" +
+                "define stream inputStream (a string, b int, c float, d long, e double, f bool); " +
+                "" +
+                "@sink(type='inMemory', topic='WSO2', @map(type='binary')) " +
+                "define stream outputStream (a string, b int, c float, d long, e double, f bool);";
+        String query = ("@info(name = 'query1') " +
+                "from inputStream " +
+                "select *  " +
+                "insert into outputStream;");
+
+        SiddhiAppRuntime siddhiAppRuntime = siddhiManager.createSiddhiAppRuntime(
+                inStreamDefinition + query);
 
+        Attribute.Type[] types = EventDefinitionConverterUtil.generateAttributeTypeArray(siddhiAppRuntime
+                .getStreamDefinitionMap().get("outputStream").getAttributeList());
+
+        InMemoryBroker.Subscriber subscriber = new InMemoryBroker.Subscriber() {
+            @Override
+            public void onMessage(Object o) {
+                Event[] events = SiddhiEventConverter.toConvertToSiddhiEvents(ByteBuffer.wrap(((ByteBuffer) o).array
+                        ()), types);
+                EventPrinter.print(events);
+                eventArrived = true;
+                for (Event event : events) {
+                    count++;
+                    switch (count) {
+                    case 1:
+                        AssertJUnit.assertEquals("test", event.getData(0));
+                        break;
+                    case 2:
+                        AssertJUnit.assertEquals("test1", event.getData(0));
+                        break;
+                    case 3:
+                        AssertJUnit.assertEquals("test2", event.getData(0));
+                        break;
+                    default:
+                        AssertJUnit.fail();
+                    }
+                }
+            }
+
+            @Override
+            public String getTopic() {
+                return "WSO2";
+            }
+        };
+        InMemoryBroker.subscribe(subscriber);
+
+        InputHandler inputHandler = siddhiAppRuntime.getInputHandler("inputStream");
+        siddhiAppRuntime.start();
+        inputHandler.send(new Object[]{"test", 36, 3.0f, 380L, 23.0, true});
+        inputHandler.send(new Object[]{"test1", 361, 31.0f, 3801L, 231.0, false});
+        inputHandler.send(new Object[]{"test2", 362, 32.0f, 3802L, 232.0, true});
+
+        Thread.sleep(300);
```

Comment:
Can't we use SiddhiTestHelper.waitForEvents and avoid Thread.sleep ?

Reply:
Yes, fixed in e65fd1e